src/file/decrypt.rs

```diff
@@ -160,20 +163,21 @@ fn get_writer_and_path(out_path: PathBuf) -> Result<(Box<dyn Write>, String), St
             out_path.display()
         ))?
     };
-    Ok((out_writer, out_logged_path.to_string()))
+    Ok(out_logged_path.to_string())
 }
 
 fn decrypt_file(
     sdk: &BlockingIronOxide,
     encrypted_document: Vec<u8>,
-    mut decrypted_writer: Box<dyn Write>,
+    out_path: &PathBuf,
     delete: bool,
     input_path: Option<&PathBuf>,
 ) -> Result<(), String> {
     let decrypt_result = sdk
         .document_decrypt(&encrypted_document)
         .map_err(|e| format!("Failed to decrypt encrypted document: {e}"))?;
     let decrypted_document = decrypt_result.decrypted_data();
+    let mut decrypted_writer = get_output_writer(out_path.clone())?;
```

Comment:
Since this function needs an owned `out_path`, it would be more clear for the function itself to take an owned value instead of a borrow. I don't think it actually makes a difference in this case though, since you'll end up cloning at the call site.
I would also probably rearrange the parameters slightly so that out_path and input_path are next to each other